client/cmd/dexc-desktop/app_darwin.go

```diff
@@ -0,0 +1,421 @@
+//go:build darwin
+
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"os/signal"
+	"path/filepath"
+	"runtime"
+	"runtime/debug"
+	"runtime/pprof"
+	"strings"
+	"sync"
+	"syscall"
+	"time"
+
+	"decred.org/dcrdex/client/app"
+	"decred.org/dcrdex/client/asset"
+	_ "decred.org/dcrdex/client/asset/bch"  // register bch asset
+	_ "decred.org/dcrdex/client/asset/btc"  // register btc asset
+	_ "decred.org/dcrdex/client/asset/dcr"  // register dcr asset
+	_ "decred.org/dcrdex/client/asset/dgb"  // register dgb asset
+	_ "decred.org/dcrdex/client/asset/doge" // register doge asset
+	_ "decred.org/dcrdex/client/asset/firo" // register firo asset
+	_ "decred.org/dcrdex/client/asset/ltc"  // register ltc asset
+	_ "decred.org/dcrdex/client/asset/zec"  // register zec asset
+	dexCore "decred.org/dcrdex/client/core"
+	"decred.org/dcrdex/client/mm"
+	"decred.org/dcrdex/client/rpcserver"
+	"decred.org/dcrdex/client/webserver"
+	"decred.org/dcrdex/dex"
+	"github.com/pkg/browser"
+	"github.com/progrium/macdriver/cocoa"
+	"github.com/progrium/macdriver/core"
+	"github.com/progrium/macdriver/objc"
+	"github.com/progrium/macdriver/webkit"
+	// Ethereum loaded in client/app/importlgpl.go
+)
+
+var (
+	webviewConfig  = webkit.WKWebViewConfiguration_New()
+	width, height  = defaultWindowWidthAndHeight()
+	maxOpenWindows = 5 // what would they want to do with more than 5 😂?
+	nOpenWindows   int
+)
+
+func init() {
+	runtime.LockOSThread()
+
+	// Set webview preferences.
+	webviewConfig.Preferences().SetValueForKey(core.True, core.String("developerExtrasEnabled"))
+	webviewConfig.Preferences().SetValueForKey(core.True, core.String("javaScriptCanAccessClipboard"))
+	webviewConfig.Preferences().SetValueForKey(core.True, core.String("DOMPasteAllowed"))
+}
+
+// hasOpenWindows is a convenience function to tell if there are any windows
+// currently open.
+func hasOpenWindows() bool {
+	return nOpenWindows > 0
+}
+
+// createNewAppWindow creates a new window with the specified URL.
+func createNewAppWindow(url string) {
+	if nOpenWindows > maxOpenWindows-1 {
+		log.Debugf("Ignoring open new window request, max number of (%d) open windows exceeded", maxOpenWindows)
+		return
+	}
+	nOpenWindows++
+
+	// Create a new webview and load the provided url.
+	webView := webkit.WKWebView_Init(core.Rect(0, 0, float64(width), float64(height)), webviewConfig)
+	req := core.NSURLRequest_Init(core.URL(url))
+	webView.LoadRequest(req)
+
+	// Create a new window and set the webview as its content view.
+	win := cocoa.NSWindow_Init(core.NSMakeRect(0, 0, 1440, 900), cocoa.NSClosableWindowMask|cocoa.NSTitledWindowMask|cocoa.NSResizableWindowMask|cocoa.NSFullSizeContentViewWindowMask|cocoa.NSMiniaturizableWindowMask, cocoa.NSBackingStoreBuffered, false)
+	win.SetTitle(fmt.Sprintf("Decred DEX Client %d", nOpenWindows))
```

Comment:
If you have 2 open, close number 1, then open another, then it would say number 2 on both.